path_finding.asciidoc

```diff
@@ -122,3 +127,21 @@ These three activities can be repeated in a _payment round_ if we use the failur
 In the next sections we will look at each of these steps in more detail, as well as more advanced payment strategies.
 
 === Channel graph construction
+
+In <<gossip>> we explain the three main messages that nodes "gossip": +node_announcement+, +channel_announcement+, and +channel_update+. These three messages allow any node to gradually construct a "map" of the Lightning Network in the form of a _channel graph_. Each of these messages provides a critical piece of information for the channel graph:
+
+node_announcement:: This contains the information about a node on the Lightning Network, such as its node ID (public key), network address (e.g. IPv4/6 or Tor), capabilities/features etc.
+
+channel_announcement:: This contains the capacity and channel ID of a public (announced) channel between two nodes and proof of the channel's existence and ownership.
+
+channel_update:: This contains a node's fee and timelock (CLTV) expectations for routing an outgoing (from that node's perspective) payment over a specified channel.
+
+In terms of a mathematical graph, the node_announcement is the information needed to create the nodes or _vertices_ of the graph. The channel_announcement allows us to create the _edges_ of the graph representing the payment channels. Since each direction of the payment channel has its own balance, we create a directed graph. The channel_update allows us to incorporate fees and timelocks to set the _cost_ or _weight_ of the graph edges.
+
+Depending on the algorithm we will use for path finding, we may establish a number of different cost functions for the edges of the graph.
```

Comment:
I think in all cases we use the same algorithm it really is just the cost function that changes.

Proposal: 

"Depending on our optimization goal for transporting the funds, we may establish a number of different cost functions (weights) for the edges of the graph"

we could append the sentence with "to be used in the path finding or min cost flow algorithm" (however I think that would be too much detail for now).